### PR TITLE
Consolidate admin required fields

### PR DESCRIPTION
**Pull request by xtomas**
* add star char for all required fields
* add validation message for required input field like in TitlePart
* consolidate color for start and validation error messages
* consolidate error message texts
* remove client validation from inputs because of inconsistent form behaviour dependent on used browser, selected locale, ...

#15434

New test content item form with required fields

image

And the same content item after Publish

image

### CODE REVIEW

**Comment by xtomas** [comment]
@dotnet-policy-service agree

**Review by sebastienros — APPROVED** [review]

**Comment by hishamco** [comment]
@all-contributors please add @xtomas for code.

File: src/OrchardCore.Themes/TheAdmin/wwwroot/css/TheAdmin.css [hunk 1]
```diff
@@ -839,6 +840,11 @@ label .field-validation-error::before {
   content: "- ";
 }
 
+label.input-required:after {
```
Comment (Skrypt) [inline src/OrchardCore.Themes/TheAdmin/wwwroot/css/TheAdmin.css:843]: let me take a look but I'm pretty sure we have something already like this.

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Views/DateField.Edit.cshtml [hunk 2]
```diff
@@ -6,11 +6,12 @@
 
 <script asp-name="jQuery-ui-i18n" at="Foot"></script>
 <div class="@Orchard.GetFieldWrapperClasses(Model.PartFieldDefinition)" id="@Html.IdFor(x => x.Value)_FieldWrapper">
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Views/DateField.Edit.cshtml:8]: Have you tried with `asp-validation-for="Value"` on this div to set the entire "field" to be set has having a validation error?

**Comment by hishamco** [comment]
@Skrypt do you have anything else to add or shall we merge this?

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Views/DateField.Edit.cshtml (hunk 2 above)
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.ContentFields/Views/DateField.Edit.cshtml:8]: Example: 

image

**Comment by Skrypt** [comment]
It's approved; you can merge but I'm just asking if we should not also add the asp-validation-class-for too then to make this consistent everywhere.

**Comment by hishamco** [comment]
What's the wrong with `asp-validation`?

**Comment by Skrypt** [comment]
It is missing everywhere for these fields. But we use it in ActivityDisplayDrivers editors for example.

**Pull request merged**